.github/workflows/pypi_deploy.yml

```diff
@@ -0,0 +1,72 @@
+# This workflow will upload a Python Package using the PyPi action workflow
+
+name: Upload Python Package
+
+on:
+  push:
+    # only run when a new tag is pushed; and only when pushed to master
+    tags:
+      - '*'    
+    branches: [ master ]
+
+jobs:
+  test:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+       py_version: ["3.6", "3.7", "3.8", "3.6-dev", "3.7-dev", "3.8-dev"] 
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.py_version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.py_version }}
+    - name: Install dependencies
+      run: |
+        pip install --upgrade setuptools pip
+        pip install --upgrade .
+        pip install -r requirements-tests.txt
+        
+    - name: Run pytest with codecoverage
+      run: |
+        coverage run --source=polyply $(which pytest) -vv polyply
+        coverage report --omit='*/bin/pytest'
+        codecov
```

Comment:
It's too bad you duplicate the testing declaration here, but I guess there's no other way?

Reply:
No you cannot have interdependent workflows in the sense that deploy waits for testing to be completed. One can in principle try to pick up tags in the other workflow and then do the deploy, but this seems safest. Otherwise you might end up realasing every commit that is pushed to a PR. And that would be bad.

Reply:
> And that would be bad.

Indeed. Leave it as is then